ports/geckolib/string_cache/lib.rs

```diff
@@ -214,7 +223,7 @@ impl fmt::Debug for Atom {
 
 impl fmt::Display for Atom {
     fn fmt(&self, w: &mut fmt::Formatter) -> fmt::Result {
-        for c in char::decode_utf16(self.as_slice().iter().cloned()) {
+        for c in self.chars() {
```

Comment:
I’ve removed `PartialEq<str> for Atom` because, I agree, it’s easy to accidentally use it instead of fast atom comparison.

But consuming `Atom::chars` is O(n) like `str::chars`, and calling it has to be explicit, so I don’t see what the problem is. When possible, it’s better than using `with_str` which allocates.